src/atom-environment.coffee

```diff
@@ -229,6 +229,10 @@ class AtomEnvironment extends Model
 
     checkPortableHomeWritable()
 
+    # Patch the `process.env` on startup to fix the problem first documented
+    # in #4126
+    process.env = @project.getEnv()
```

Comment:
Looking deeper, it appears that packages aren't loaded until `AtomEnvironment.startEditorWindow`:

https://github.com/atom/atom/blob/ld-osx-environment/src/atom-environment.coffee#L674

So doing the patching of the environment in the `AtomEnvironment` constructor does make it happen before packages are loaded.